tests/Unit/NumericalAlgorithms/Lapack/Test_GeneralizedEigenvalue.cpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "NumericalAlgorithms/Lapack/GeneralizedEigenvalue.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+namespace {
+void test_generalized_eigenvalue() {
+  // Set up some test matrices
+  Matrix A(2, 2, 0.0);
+  Matrix B(2, 2, 0.0);
+  A(0, 0) = 1.0;
+  A(0, 1) = 2.0;
+  A(1, 0) = -3.0;
+  A(1, 1) = -4.0;
+  B(0, 0) = 4.0;
+  B(0, 1) = -3.0;
+  B(1, 0) = -2.0;
+  B(1, 1) = 1.0;
+
+  // Set up vectors and a matrix to store the eigenvalues and eigenvectors
+  DataVector eigenvalues_real_part(2, 0.0);
+  DataVector eigenvalues_im_part(2, 0.0);
+  Matrix eigenvectors(2, 2, 0.0);
+
+  // Get the generalized eigenvectors and eigenvalues
+  generalized_eigenvalue(A, B, eigenvalues_real_part, eigenvalues_im_part,
+                         eigenvectors);
+
+  // Expected eigenvalues and eigenvectors  Matrix expected_vectors(2, 2, 0.0);
+  // Eigenvectors normalized so last component is 1
+  constexpr double expected_larger_eigenvalue = 10.099019513592784;
+  constexpr double expected_smaller_eigenvalue = -0.09901951359278449;
+  constexpr double expected_larger_eigenvector_value = 0.819803902718557;
+  constexpr double expected_smaller_eigenvector_value = -1.2198039027185574;
+
+  constexpr double eps = 1.e-14;
+
+  // Check eigenvalues
+  CHECK(abs(eigenvalues_im_part[0]) < eps);
+  CHECK(abs(eigenvalues_im_part[1]) < eps);
+  CHECK(abs(max(eigenvalues_real_part) - expected_larger_eigenvalue) < eps);
+  CHECK(abs(min(eigenvalues_real_part) - expected_smaller_eigenvalue) < eps);
```

Comment:
These checks would be clearer if you used `approx`. So 
```cpp
CHECK(eigenvalues_im_part[0] == approx(0.)); 
... 
CHECK(max(eigenvalues_real_part) == approx(expected_larger_eigenvalue)));
...
```
I think approx by default checks to `1.e-14`